src/controllers/source.ts

```diff
@@ -4,13 +4,27 @@ import { MiscService } from '../services/misc';
 import { SourcePostTransformationService } from '../services/source/postTransformation';
 import { ControllerUtility } from './util';
 import logger from '../logger';
+import { FixMe } from '../util/types';
 
 export class SourceController {
+  private static enrichMetadata(
+    metadata: { namespace: string; cluster: string; features: FixMe },
+    source: string,
+    version: string,
+  ) {
+    return {
+      ...metadata,
+      source,
+      version,
+    };
+  }
+
   public static async sourceTransform(ctx: Context) {
     logger.debug('Native(Source-Transform):: Request to transformer::', ctx.request.body);
     const requestMetadata = MiscService.getRequestMetadata(ctx);
     const events = ctx.request.body as object[];
     const { version, source }: { version: string; source: string } = ctx.params;
+    const enrichedMetadata = this.enrichMetadata(requestMetadata, source, version);
```

Comment:
do we really need a function here ?
```suggestion
    const enrichedMetadata = {
      ...metadata,
      source,
      version,
    };
```

Reply:
Done